### PR TITLE
Restore persisted state after data is fully loaded

### PR DESCRIPTION
**Pull request by marinaaisa**
Bug/issue #101539534, if applicable: 

## Summary

Restore persisted state after data is fully loaded

## Dependencies

NA

## Testing

Steps:

1. Go deeply into the tree
2. Refresh the page
3. Assert the page is properly highlighted

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by mportiz08** [comment]
A deeply nested item is being highlighted as expected now when loading the URL directly or after refreshing.

However, I'm noticing that the same issue is still happening for other items, like an article at the top level of the navigation tree.

Maybe there are additional changes to make to ensure items are always being highlighted as expected (don't have any obvious suggestions at the moment, sorry)? I tested with the DocC docs as an example.

**Comment by marinaaisa** [comment]
> However, I'm noticing that the same issue is still happening for other items, like an article at the top level of the navigation tree.

You were right, @mportiz08 . I took another approach to load the skeleton screen in the Navigator instead of NavigatorCard component. To keep the same UI as we had before it took me a big refactor, but it's working quite well.
Please, review again. Thanks!

**Comment by marinaaisa** [comment]
@swift-ci test

File: src/components/Navigator/BaseNavigatorCard.vue [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+            <button
+              aria-label="Close documentation navigator"
+              :id="SIDEBAR_HIDE_BUTTON_ID"
+              class="close-card"
+              :class="{ 'hide-on-large': !allowHiding }"
+              @click="handleHideClick"
+            >
+              <SidenavIcon class="icon-inline close-icon" />
+            </button>
+            <slot name="head" className="navigator-head" id="INDEX_ROOT_KEY" />
```
Comment (dobromir-hristov) [inline src/components/Navigator/BaseNavigatorCard.vue:26]:
```suggestion
            <slot name="head" className="navigator-head" :id="INDEX_ROOT_KEY" />
```

File: src/components/Navigator/BaseNavigatorCard.vue [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+          </div>
+        </div>
+        <slot name="post-head" />
+        <slot name="body" className="card-body"/>
+        <slot name="filter"/>
```
Comment (dobromir-hristov) [inline src/components/Navigator/BaseNavigatorCard.vue:31]: we dont need 3 slots one after the other. One slot is fine imho

File: src/components/Navigator/NavigatorCard.vue [hunk 3]
```diff
@@ -9,126 +9,111 @@
...
-    <div class="navigator-card-full-height">
-      <div class="navigator-card-inner">
-        <div class="head-wrapper">
-          <div class="head-inner">
-            <button
-              aria-label="Close documentation navigator"
-              :id="SIDEBAR_HIDE_BUTTON_ID"
-              class="close-card"
-              :class="{ 'hide-on-large': !allowHiding }"
-              @click="handleHideClick"
```
Comment (dobromir-hristov) [inline src/components/Navigator/NavigatorCard.vue:22]: `handleHideClick` this is not removed from the methods, but it was copied to the base card

File: src/components/Navigator/BaseNavigatorCard.vue [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        trigger.focus();
+      }
+    },
+  },
+};
+</script>
+
+<style scoped lang='scss'>
+@import 'docc-render/styles/_core.scss';
+@import '~vue-virtual-scroller/dist/vue-virtual-scroller.css';
```
Comment (dobromir-hristov) [inline src/components/Navigator/BaseNavigatorCard.vue:77]: Why import it here, when its not used here?

**Review by dobromir-hristov — COMMENTED** [review]
LGTM but there are a few small remarks.

File: src/components/Navigator/BaseNavigatorCard.vue (hunk 2 above)
Comment (marinaaisa) [reply src/components/Navigator/BaseNavigatorCard.vue:31]: We may want to separate post-head content from body content, to make sure we don't overwrite the body when adding a post-head.

But yes, body and filter can be the same. I changed it.

**Comment by marinaaisa** [comment]
@swift-ci test

**Comment by marinaaisa** [comment]
@swift-ci test

File: src/components/Navigator/NavigatorCard.vue [hunk 5]
```diff
@@ -10,23 +10,15 @@
...
-        @click.alt.native.prevent="toggleAllNodes"
-      >
-        <h2 class="card-link">
-          {{ technology }}
-        </h2>
-        <Badge v-if="isTechnologyBeta" variant="beta" />
-      </Reference>
-    </template>
     <template #post-head>
       <slot name="post-head" />
```
Comment (dobromir-hristov) [inline src/components/Navigator/NavigatorCard.vue:23]: You could theoretically put this inside the `#body` slot, right?
```vue
<template #body="{ className }">
    <slot name="post-head" />
     <div> bla bla other stuff
```

**Review by dobromir-hristov — COMMENTED** [review]
Left a few small comments. The rest I think is fine and it works fine too.

File: src/components/Navigator/BaseNavigatorCard.vue [hunk 6]
```diff
@@ -23,12 +23,21 @@
             >
               <SidenavIcon class="icon-inline close-icon" />
             </button>
-            <slot name="head" className="navigator-head" id="INDEX_ROOT_KEY" />
+            <Reference
+              :url="technologyPath"
+              class="navigator-head"
+              :id="INDEX_ROOT_KEY"
+              @click.alt.native.prevent="clickReferenceAltEvent"
```
[the remark this answers is not in the document]
Comment (marinaaisa) [reply src/components/Navigator/BaseNavigatorCard.vue]: oh god, true, why did I make it so difficult? lol

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Review by dobromir-hristov — APPROVED** [review]

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Pull request merged**